spark/src/main/scala/org/apache/comet/parquet/ParquetFilters.scala

```diff
@@ -723,20 +723,22 @@ class ParquetFilters(
           .lift(nameToParquetField(name).fieldType)
           .map(_(nameToParquetField(name).fieldNames, value))
 
-      case sources.LessThan(name, value) if canMakeFilterOn(name, value) =>
+      case sources.LessThan(name, value) if (value != null) && canMakeFilterOn(name, value) =>
```

Comment:
why do we need to change Spark's test?

Reply:
This is not a test code

This file was copied from Spark for modification. Then there is a logic update for SPARK-47120
https://github.com/apache/spark/pull/45202/files